src/averageAges.js

```diff
@@ -52,7 +73,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const female = people.filter(woman =>
+    woman.sex === 'f'
+    && people.some(person => person.mother === woman.name));
+
+  const averageAges = [];
+  let children;
+
+  onlyWithSon !== undefined
+    ? children = people.filter(person => person.sex === 'm'
+      && female.map(woman => person.mother === woman.name))
+    : children = people.filter(person =>
+      female.map(woman => person.mother === woman.name));
+
+  children.map(child => female.map(woman => child.mother === woman.name
+    ? averageAges.push(child.born - woman.born) : child));
```

Comment:
Try not to use 'push' in this task. You can use the method 'map' to remap the array of 'children' and find the age difference. Inside the 'map' method you can use 'find' method, to find the exact mother of the current child to find her birth year and calculate the difference. Then you will receive the array of differences